modules/currency.js

```diff
@@ -0,0 +1,220 @@
+
+import bidfactory from 'src/bidfactory';
+import { STATUS } from 'src/constants';
+import { ajax } from 'src/ajax';
+import * as utils from 'src/utils';
+import bidmanager from 'src/bidmanager';
+
+$$PREBID_GLOBAL$$.currency = setConfig;
+
+const DEFAULT_CURRENCY_RATE_URL = 'http://currency.prebid.org/latest.json';
+const CURRENCY_RATE_PRECISION = 4;
+
+var bidResponseQueue = [];
+var conversionCache = {};
+var currencyRatesLoaded = false;
+var adServerCurrency = 'USD';
+
+// Used as reference to the original bidmanager.addBidResponse
+var originalBidResponse;
+
+export var currencySupportEnabled = false;
+export var currencyRates = {};
+var currencyOverrides = {};
+
+export function setConfig(config) {
+  var url = DEFAULT_CURRENCY_RATE_URL;
+  if (typeof config.adServerCurrency === 'string') {
+    utils.logInfo('enabling currency support', arguments);
+
+    adServerCurrency = config.adServerCurrency;
+    if (config.conversionRateFile) {
+      utils.logInfo('currency using override conversionRateFile:', config.conversionRateFile);
+      url = config.conversionRateFile;
+    }
+    initCurrency(url);
+  } else {
+    // currency support is disabled, setting defaults
+    utils.logInfo('disabling currency support');
+    resetCurrency();
+  }
+  if (typeof config.currencyOverrides === 'object') {
+    currencyOverrides = config.currencyOverrides;
+  }
+}
+
+function initCurrency(url) {
+  conversionCache = {};
+  currencySupportEnabled = true;
+
+  if (!originalBidResponse) {
+    utils.logInfo('Installing addBidResponse decorator for currency module', arguments);
+
+    originalBidResponse = bidmanager.addBidResponse;
+    bidmanager.addBidResponse = addBidResponseDecorator(bidmanager.addBidResponse);
```

Comment:
Thinking over the issue @snapwich brought up [here](https://github.com/prebid/Prebid.js/pull/1421#issuecomment-319441973): I don't think we should support modules which monkey-patch functions in core.

Reason being: I've heard that we expect other people to write their own external modules. If so, then this behavior means that all of Prebid's _internal_ APIs are now _external_ APIs. If we remove a method or change a signature on anything we export, they could reasonably claim that it's a breaking change to the library.

I propose that we deliberately define all the "plugin points" in prebid-core, which are functions which we expect modules to call. For example, `registerBidAdapter()` is a plugin point today.

@snapwich @mkendall07 What do you guys think?

I'm not sure what the right "plugin point" API would be for this feature... but I just think it should be explicit. Off the top of my head, something like `addBidPostprocessor((bid) => bid))` looks reasonable--although I'm not sure it fits with the spirit of the project.